### DISCUSSION TITLE
Autostart under ubuntu 16.04 not working

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#846] [Installation issue: custom]
hello.

First, thanks for your work. I really like searx.

Here is my problem. I'm not able to start searx at boot time. I have to login to the VM after the VM is started and have to restart the uwsgi service. If I don't do that searx is not starting. Logs says:

```
Sat Jan 28 20:14:38 2017 - added /usr/local/searx/ to pythonpath.
ERROR:searx.engines:Cannot load engine "soundcloud"
Traceback (most recent call last):
  File "/usr/local/searx/searx/engines/__init__.py", line 63, in load_engine
    engine = load_module(engine_module + '.py', engine_dir)
  File "/usr/local/searx/searx/utils.py", line 300, in load_module
    module = load_source(modname, filepath)
  File "/usr/local/searx/searx/engines/soundcloud.py", line 62, in <module>
    guest_client_id = get_client_id()
  File "/usr/local/searx/searx/engines/soundcloud.py", line 41, in get_client_id
    response = http_get("https://soundcloud.com")
  File "/usr/local/searx/searx/poolrequests.py", line 86, in get
    return request('get', url, **kwargs)
  File "/usr/local/searx/searx/poolrequests.py", line 79, in request
    response = session.request(method=method, url=url, **kwargs)
  File "/usr/local/searx/searx-ve/local/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/searx/searx-ve/local/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/searx/searx-ve/local/lib/python2.7/site-packages/requests/adapters.py", line 497, in send
    raise SSLError(e, request=request)
SSLError: ("bad handshake: Error([('', 'osrandom_rand_bytes', 'getrandom() initialization failed.')],)",)
Traceback (most recent call last):
  File "/usr/local/searx/searx/webapp.py", line 109, in <module>
    initialize_engines(settings['engines'])
  File "/usr/local/searx/searx/engines/__init__.py", line 221, in initialize_engines
    engines[engine.name] = engine
AttributeError: 'NoneType' object has no attribute 'name'
```

Any hints on that? "band handshake" seems to be the main reason.

thanks and cheers
t.

**Comment by dalf** [comment]
Can you check if pyopenssl, pyasn1, ndg-httpsclient are installed ? 
If you use a virtualenv, can you check that the the virtualenv is setup correctly on the boot ?

**Comment by ghost** [comment]
@dalf I guess pyopenssl, pyasn1, ndg-httpsclient are installed properly otherwise searx would not work at all, no? And for the second question. All the machine is doing is starting uwsgi service which is:

```
[uwsgi]
# Who will run the code
uid = searx
gid = searx

# disable logging for privacy
disable-logging = true

# Number of workers (usually CPU count)
workers = 4

# The right granted on the created socket
chmod-socket = 666

# Plugin to use and interpretor config
single-interpreter = true
master = true
plugin = python

# Module to import
module = searx.webapp

# Virtualenv and python path
virtualenv = /usr/local/searx/searx-ve/
pythonpath = /usr/local/searx/
chdir = /usr/local/searx/searx/
```

thanks and cheers

**Comment by dalf** [comment]
without pyopenssl, pyasn1, ndg-httpsclient, searx seems to work but some engines may not work because the HTTPS connection can't be made (there are optional dependencies of the requests module).

> I have to login to the VM after the VM is started and have to restart the uwsgi service. If I don't do that searx is not starting. 

To clarify : it's mean the error doesn't appear after restarting uswgi ? If yes, it's an Linux distribution problem.  

```getrandom() initialization failed``` is the real error, it's an urllib3 error.

**Comment by ghost** [comment]
@dalf Thanks for your answer. Problem is gone with searx 0.11 or one of the ubuntu updates. Now searx starts fine. Maybe you are right and it was an ubuntu problem.

cheers
